x/gov/codec.proto

```diff
@@ -89,29 +100,43 @@ message TextProposal {
   TallyResult vote_state = 10 [(gogoproto.nullable) = false];
   enum Status {
     // An empty value is invalid and not allowed
-    TEXT_PROPOSAL_STATUS_INVALID = 0 [(gogoproto.enumvalue_customname) = "Invalid"];
+    PROPOSAL_STATUS_INVALID = 0 [(gogoproto.enumvalue_customname) = "Invalid"];
     // Initial status of a proposal when persisted.
-    TEXT_PROPOSAL_STATUS_SUBMITTED = 1 [(gogoproto.enumvalue_customname) = "Submitted"];
+    PROPOSAL_STATUS_SUBMITTED = 1 [(gogoproto.enumvalue_customname) = "Submitted"];
     // Final status of a proposal when the tally was executed
-    TEXT_PROPOSAL_STATUS_CLOSED = 2 [(gogoproto.enumvalue_customname) = "Closed"];
+    PROPOSAL_STATUS_CLOSED = 2 [(gogoproto.enumvalue_customname) = "Closed"];
     // A proposal can be deleted before the voting start time by the owner. When this happens the final status
     // is Withdrawn.
-    TEXT_PROPOSAL_STATUS_WITHDRAWN = 3 [(gogoproto.enumvalue_customname) = "Withdrawn"];
+    PROPOSAL_STATUS_WITHDRAWN = 3 [(gogoproto.enumvalue_customname) = "Withdrawn"];
   }
   // Status represents the high level position in the life cycle of the proposal. Initial value is submitted.
   Status status = 11;
   enum Result {
     // An empty value is invalid and not allowed
-    TEXT_PROPOSAL_RESULT_INVALID = 0 [(gogoproto.enumvalue_customname) = "Empty"];
+    PROPOSAL_RESULT_INVALID = 0 [(gogoproto.enumvalue_customname) = "Empty"];
     // Until a final tally has happened the status is undefined
-    TEXT_PROPOSAL_RESULT_UNDEFINED = 1 [(gogoproto.enumvalue_customname) = "Undefined"];
+    PROPOSAL_RESULT_UNDEFINED = 1 [(gogoproto.enumvalue_customname) = "Undefined"];
     // Final result of the tally
-    TEXT_PROPOSAL_RESULT_ACCEPTED = 2 [(gogoproto.enumvalue_customname) = "Accepted"];
+    PROPOSAL_RESULT_ACCEPTED = 2 [(gogoproto.enumvalue_customname) = "Accepted"];
     // Final result of the tally
-    TEXT_PROPOSAL_RESULT_REJECTED = 3 [(gogoproto.enumvalue_customname) = "Rejected"];
+    PROPOSAL_RESULT_REJECTED = 3 [(gogoproto.enumvalue_customname) = "Rejected"];
   }
   // Result is the final result based on the votes and election rule. Initial value is Undefined.
   Result result = 12;
+  enum Type {
```

Comment:
So there is an enum and a oneof below that must match.
Why is the enum there? Just for reability for those who don't want to parse the other?

Reply:
As you figured out already this enum type is to map proposal types with executors that work with the proposal details.